### ISSUE TITLE
Feature Request: var by

### ISSUE THREAD

**Issue by xtrm0** [dafny-lang/dafny#1713]
In cases like the following, we first need to prove the existence of x:
```dafny
var x :| Property(x);
```
Which usually creates code like this:
```dafny
assert exists x :: Property(x)
by {
  ...proof...
  assert exists x :: Property(x);
}
var x :| Property(x);
```

It would be nice to have this syntathic sugar that would compact both into a single statement, specially nice if Property isn't a single predicate.
```dafny
var x :| Property(x)
by {
  ...proof...
  assert exists x :: Property(x);
}
```

**Comment by MikaelMayer** [comment]
Would this work for now?

    var x :| lemma_assert_existence(); Property(x)

That would still force to write a second lemma, but at least you have everything on one line.

**Comment by xtrm0** [comment]
The point is also avoiding redundant text on statements that are only used in a single lemma.

I am currently using a preprocessor macro that does your solution, but it would be nice to have the feature directly in the dafny parser, it can probably be done directly in the .atg.

**Comment by keyboardDrummer** [comment]
I think this was resolved in Dafny 4.9, `by` clauses are now available on `var` statements as well: https://github.com/dafny-lang/dafny/releases/tag/v4.9.0
